SYCL/KernelAndProgram/kernel-bundle-get-kernel.cpp

```diff
@@ -0,0 +1,51 @@
+// RUN: %clangxx -fsycl -fsycl-targets=%sycl_triple %s -o %t.out
+// RUN: %CPU_RUN_PLACEHOLDER %t.out
+// RUN: %GPU_RUN_PLACEHOLDER %t.out
+// RUN: %ACC_RUN_PLACEHOLDER %t.out
+
+// This test checks whether the Kernel object returned from
+// kernel_bundle::get_kernel<typename KernelName>() is the same as a Kernel
+// object retrieved via other methods.
+
+#include <sycl/sycl.hpp>
+
+class KernelA;
+
+int main() {
+  sycl::queue Queue;
+
+  sycl::device Dev = Queue.get_device();
+
+  sycl::context Ctx = Queue.get_context();
+
+  Queue.submit([&](sycl::handler &CGH) { CGH.single_task<KernelA>([=]() {}); });
+
+  sycl::kernel_bundle KernelBundle =
+      sycl::get_kernel_bundle<sycl::bundle_state::executable>(Ctx, {Dev});
+
+  auto FoundKernel = KernelBundle.get_kernel<KernelA>();
+  auto ExpectedKernel = KernelBundle.get_kernel(sycl::get_kernel_id<KernelA>());
+
+  // operator== isn't guaranteed to work in this scenario, so compare traits
+  // about the kernels instead.
+  auto FoundKernelName =
+      FoundKernel.get_info<sycl::info::kernel::function_name>();
+  auto ExpectedKernelName =
+      ExpectedKernel.get_info<sycl::info::kernel::function_name>();
+
+  auto FoundKernelNumArgs =
+      FoundKernel.get_info<sycl::info::kernel::num_args>();
+  auto ExpectedKernelNumArgs =
+      ExpectedKernel.get_info<sycl::info::kernel::num_args>();
+
+  auto FoundKernelContext = FoundKernel.get_info<sycl::info::kernel::context>();
+  auto ExpectedKernelContext =
+      ExpectedKernel.get_info<sycl::info::kernel::context>();
+
+  assert(FoundKernelName == ExpectedKernelName);
+  assert(FoundKernelNumArgs == ExpectedKernelNumArgs);
+  assert(FoundKernelContext == ExpectedKernelContext);
+  assert(FoundKernel.get_kernel_bundle() == ExpectedKernel.get_kernel_bundle());
```

Comment:
Does the spec guarantee this to be the case? I couldn't find anything when I did a quick skim, but maybe I missed it.

Reply:
I'm not sure that it does, it was the only way that I could think to compare them for equality considering operator== isn't guaranteed to work here. I'm not sure how else to do this?

Reply:
Good question. I suppose you could check that they have the same kernels, but I don't think that's even a guarantee either. I think we can live without this last check, relying on the other checks.